addons/assignGear/functions/fnc_assignGearVehicle.sqf

```diff
@@ -90,36 +94,276 @@ private _transportMagazines = getArray(_path >> "TransportMagazines");
+        } count _transportMagazines; // count used here for speed, make sure nil is above this line
+
+        // transportItems
+        {
+            (_x splitString ":") params ["_classname", ["_amount", "1", [""]]];
+            _availableItems pushBackUnique _classname;
+            private _config = configFile >> "CfgWeapons" >> _classname;
+            (_loadoutInfo getOrDefault [_classname, [0, _config call _getMassLbs, getText (_config >> "displayName")]]) params ["_existingAmount", "_mass", "_name"];
+            _loadoutInfo set [_classname, [_existingAmount + parseNumber _amount, _mass, _name]];
+            nil
+        } count _transportItems; // count used here for speed, make sure nil is above this line
+
+        // transportWeapons
+        {
+            (_x splitString ":") params ["_classname", ["_amount", "1", [""]]];
+            private _disposableName = [cba_disposable_LoadedLaunchers, _classname, "get", ""] call FUNC(getDisposableInfo);
+            if (_disposableName != "") then {
+                TRACE_2("cba_disposable_LoadedLaunchers replace",_classname,_disposableName);
+                _classname = _disposableName;
+            };
+            _availableWeapons pushBackUnique _classname;
+            private _config = configFile >> "CfgWeapons" >> _classname;
+            (_loadoutInfo getOrDefault [_classname, [0, _config call _getMassLbs, getText (_config >> "displayName")]]) params ["_existingAmount", "_mass", "_name"];
+            _loadoutInfo set [_classname, [_existingAmount + parseNumber _amount, _mass, _name]];
+            nil
+        } count _transportWeapons; // count used here for speed, make sure nil is above this line
+
+        // transportBackpacks
+        {
+            (_x splitString ":") params ["_classname", ["_amount", "1", [""]]];
+            _availableBackpacks pushBackUnique _classname;
+            private _config = configFile >> "CfgBackpacks" >> _classname;
+            (_loadoutInfo getOrDefault [_classname, [0, _config call _getMassLbs, getText (_config >> "displayName")]]) params ["_existingAmount", "_mass", "_name"];
+            _loadoutInfo set [_classname, [_existingAmount + parseNumber _amount, _mass, _name]];
+            nil
+        } count _transportBackpacks; // count used here for speed, make sure nil is above this line
+
+        while {
+            _availableBackpacks isNotEqualTo [] ||
+            { _availableItems isNotEqualTo [] } ||
+            { _availableWeapons isNotEqualTo [] } ||
+            { _availableMagazines isNotEqualTo [] }
+        } do {
+            scopeName "item_select_loop";
+            private _currentWeight = 0;
+            private _box = createVehicle [_transportBoxClassname, [0, 0, 0], [], 0, "NONE"];
+            clearWeaponCargoGlobal _box;
+            clearMagazineCargoGlobal _box;
+            clearItemCargoGlobal _box;
+            clearBackpackCargoGlobal _box;
+            [_box, 0] call ace_cargo_fnc_setSize;
```

Comment:
Why zero size?

Reply:
design choice: i considered increasing space in the vehicle, but if a mission maker artificially limits cargo space but still wants these boxes they should be allowed. i thought about force putting them in, but that’s poor mission flow

Reply:
Aight, sounds good to me. Only thing of note is that it changes the time to load/unload to 0 seconds (load/unload time is 5 * the setSize variable).

Reply:
that is very annoying :( if ACE changes this’ll probably update too